src/tools/optimization-options.h

```diff
@@ -26,6 +26,17 @@
 namespace wasm {
 
 struct OptimizationOptions : public ToolOptions {
+  void parse(int argc, const char* argv[]) {
+    ToolOptions::parse(argc, argv);
+
+    // After parsing the arguments, update defaults based on the optimize/shrink
+    // levels.
+    if (passOptions.optimizeLevel >= 2 || passOptions.shrinkLevel >= 1) {
```

Comment:
Didn't we make a change so that optimize levels can change over time in the case of e.g. `-O1 -O2 -O1`? Does this play well with that?

Reply:
This uses the global optimization level, so it is affected by the last of those. As StackIR only happens during binary writing, which is at the end, it seems like the natural thing here.